### ISSUE TITLE
AWS Migration [A&R]

### ISSUE THREAD

**Issue by LindsayMacfarlane** [bcgov/bcparks-ar-api#390]
#### Description:

[Description of the task]

#### Acceptance Criteria:
[Note: Use 'Given/When/Then' format if it makes sense to. Otherwise, a simple checklist that can be tested.]

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by
- Blocking

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by LindsayMacfarlane** [comment]
Followed up with Team Osprey on April 22, 2025 - From our understanding, our 'projects' need to be created by the platform team and until then we are blocked on moving forward. Assumption is that it will happen sometime in FY 2025-26. Added the blocked label to this ticket.

**Comment by marklise** [comment]
Ready for testing:

Admin: https://test-ar.bcparks.ca/
API Gateway endpoint: https://omljya6c.execute-api.ca-central-1.amazonaws.com/lza-test/

**Comment by manuji** [comment]
Tested on TEST: **Pending**
* Verified that all the features are working as expected except for the Adding a new Sub area. Unable to check if this is an existing issue or new issue after migration. 

<img width="1455" height="877" alt="Image" src="https://github.com/user-attachments/assets/4783644a-641c-4aa4-84d9-3eceff5079bd" />

FYI @marklise

**Comment by marklise** [comment]
@manuji good catch!

1. Data was migrated from prod to `lza-test`, including all subareas (highest ID: `1037`)
2. The `ConfigAr-lza-test` counter was incorrectly set to `2` (from my earlier smoke testing)
3. When trying to create a new subarea, the system would try to assign ID `3`, which would conflict with existing data causing validation issues.

Fix applied - `lastID=1037` so new subareas will start at ID `1038`.  Feel free to resume testing!

**Comment by marklise** [comment]
Production CF domain: https://d2dz2n9legxv02.cloudfront.net/

@manuji you're good to QA.

**Comment by manuji** [comment]
@marklise , while I was testing I noticed that if I add a new subArea, that subArea got the SubArea ID = 1. Not sure if we are starting the ID's from starting 1 or continuing from the last subArea id. 

<img width="465" height="51" alt="Image" src="https://github.com/user-attachments/assets/e3f3fe97-2a2f-455e-8884-f8ebc328d361" />
 
Everything else looks good to me. Thank you.

**Comment by marklise** [comment]
@manuji Thank you for the testing.  I replaced incorrect `pk=subAreaID` record in ConfigAr-lza-prod with correct value from old prod (`lastID=1037`). Next subarea will now correctly receive ID 1038.

**Comment by manuji** [comment]
Tested and confirmed that the above mentioned issue in my previous comment has been fixed and the environment is ready to use after the url changes. 
This ticket can be passed to PO review and UAT to get the final approval from the client. fyi @LindsayMacfarlane , @nicolebottoms @sallyhaynes 

New prod url: https://d2dz2n9legxv02.cloudfront.net/

Thank you very much @marklise

**Comment by LindsayMacfarlane** [comment]
<img width="617" height="215" alt="Image" src="https://github.com/user-attachments/assets/6d1147b1-6b59-4661-9232-b2558917dc72" />

**Comment by LindsayMacfarlane** [comment]
Reviewed A&R through the link provided above. While I didn't edit any data (given it is the POD environment) - the general functionality seems to work as expected. Approved for completion. 

@manuji @marklise
